actors/miner/src/lib.rs

```diff
@@ -2995,6 +2992,198 @@ impl Actor {
+            // besides, after verification, the corresponding PostProof is deleted, leaving other PostProof intact.
+            // thus it's necessary that for those moved partitions, there's an empty partition left in place to keep the partitions from re-indexing.
+            if deadline_available_for_optimistic_post_dispute(
+                policy,
+                current_deadline.period_start,
+                params.from_deadline,
+                rt.curr_epoch(),
+            ) {
+                let proofs_snapshot = from_deadline
+                    .optimistic_proofs_snapshot_amt(store)
+                    .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load proofs snapshot")?;
+
+                let partitions_snapshot =
+                    from_deadline.partitions_snapshot_amt(store).context_code(
+                        ExitCode::USR_ILLEGAL_STATE,
+                        "failed to load partitions snapshot",
+                    )?;
+
+                // Find the proving period start for the deadline in question.
+                let target_deadline =
+                    nearest_occured_deadline_info(policy, &current_deadline, params.from_deadline);
+
+                // Load sectors for the dispute.
+                let sectors = Sectors::load(rt.store(), &from_deadline.sectors_snapshot)
+                    .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load sectors array")?;
+
+                proofs_snapshot
+                    .for_each(|_, window_proof| {
+                        if !params.partitions.contains_any(&window_proof.partitions) {
+                            return Ok(());
+                        }
+
+                        let mut all_sectors =
+                            Vec::<BitField>::with_capacity(params.partitions.len() as usize);
+                        let mut all_ignored =
+                            Vec::<BitField>::with_capacity(params.partitions.len() as usize);
+
+                        for partition_idx in window_proof.partitions.iter() {
+                            let partition = partitions_snapshot
+                                .get(partition_idx)
+                                .context_code(
+                                    ExitCode::USR_ILLEGAL_STATE,
+                                    "failed to load partitions snapshot for proof",
+                                )?
+                                .ok_or_else(|| {
+                                    actor_error!(
+                                        illegal_state,
+                                        "failed to load partitions snapshot for proof"
+                                    )
+                                })?;
+                            if !partition.faults.is_empty() {
+                                return Err(actor_error!(
+                                    illegal_argument,
```

Comment:
Forbidden is probably better here and below. The argument is well-formed, just temporarily not allowed.

Reply:
Done.